product_sequence/models/product_product.py

```diff
@@ -32,7 +32,8 @@ def create(self, vals):
     @api.multi
     def write(self, vals):
         for product in self:
-            if product.default_code in [False, '/']:
+            if (product.default_code in [False, '/'] and vals.get(
+                    'default_code', '/') == '/'):
```

Comment:
This seems wrong to me.  It seems very unlikely both those conditions would be met except for the very specific behaviour the commit is for.  For example if a default_code already existed and it was removed, this test evaluates False yet in this situation we would expect a pass.  Also if a default_code did not exist, and we were writing something else, say uom_id this test also fails where it would be expected to pass. 

A much simpler way would be to call super first, then run the conditional if with a simple assignment write. e.g.
```python
result = super()
for p in self:
   if p.default_code in {False, '/'):
        <exec code>
        p.default_code = seq.next_by_id()
return result
```
there is an issue with performance, however given it is only ever going to execute that code block once ever on a product in normal usage and usually then one at a time it is acceptable IMO